pkg/domain/infra/abi/containers.go

```diff
@@ -291,15 +291,31 @@ func (ic *ContainerEngine) ContainerStop(ctx context.Context, namesOrIds []strin
 				return err
 			}
 		}
-		err = c.Cleanup(ctx)
-		if err != nil {
-			// Issue #7384 and #11384: If the container is configured for
-			// auto-removal, it might already have been removed at this point.
-			// We still need to clean up since we do not know if the other cleanup process is successful
-			if c.AutoRemove() && (errors.Is(err, define.ErrNoSuchCtr) || errors.Is(err, define.ErrCtrRemoved)) {
-				return nil
+		if c.AutoRemove() && !c.ShouldRestart(ctx) {
```

Comment:
do not call c.ShouldRestart(ctx) here, a container that is stopped by hand cannot by definition restart again.